libs/server-sdk/src/events/event_scope.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <launchdarkly/events/event_processor_interface.hpp>
+
+#include "event_factory.hpp"
+
+namespace launchdarkly::server_side {
+
+class EventScope {
```

Comment:
This is slightly different than the Rust implementation because we rely on the `NullEventProcessor` pattern in this SDK. 

There's no need for a bool to check if the processor is enabled or not, before sending the event. 

But, there is an advantage to the Rust way of doing things - no need to construct the event arguments only to throw them away. 

I'm not sure this is a real problem though, perhaps the compiler can determine that we never reassign the runtime EventProcessor object and just optimize the whole thing away.